packages/api-browser/src/window.js

```diff
@@ -26,9 +28,14 @@ class Window {
 
   close() {
     // Close only works on windows that were opened by the current window
-    if (this.innerWindow) {
-      this.innerWindow.close();
-    }
+    return new Promise((resolve, reject) => {
```

Comment:
How about refactoring to remove some of the repetition? For example:

```
const withInnerWindow(fn) =>
  new Promise((resolve, reject) => {
      if (this.innerWindow) {
        fn(this.innerWindow);
        resolve();
      } else {
        reject(windowClosedError);
      }
    });
```

Then `close` becomes the following:

```
  close() {
    return withInnerWindow(innerWindow => innerWindow.close());
  }
```